backend/src/vantracking/tracker.py

```diff
@@ -0,0 +1,136 @@
+        # the next logical stop and estimate the time to arrive to that.
+
+        stops = self.cache.get_stops(van_id)
+        current_stop_index = self.cache.get_current_stop_index(van_id)
+        # We want to consider all of the stops that are coming up for this van, as that allows to handle cases where a
+        # stop is erroneously skipped. Also make sure we include subsequent stops that wrap around. The wrap around slice
+        # needs to be bounded to 0 to prevent a negative index causing weird slicing behavior.
+        subsequent_stops = (
+            stops[current_stop_index + 1 :] + stops[: max(current_stop_index - 1, 0)]
+        )
+        locations = self.cache.get_locations(van_id)
+        for i, stop in enumerate(subsequent_stops):
+            longest_subset: list[datetime] = []
+            current_subset: list[datetime] = []
+
+            # Find the longest consequtive subset (i.e streak) where the distance of the past couple of
+            # van locations is consistently within this stop's radius.
+            for location in locations:
+                distance = _distance_m(
+                    location.coordinate,
+                    Coordinate(latitude=stop.lat, longitude=stop.lon),
+                )
+                if distance < THRESHOLD_RADIUS_M:
+                    current_subset.append(location.timestamp)
+                else:
+                    if len(current_subset) > len(longest_subset):
+                        longest_subset = current_subset
+                    current_subset = []
+            if len(current_subset) > len(longest_subset):
+                longest_subset = current_subset
+
+            if longest_subset:
+                # A streak exists, find the overall duration that this van was within the stop's radius. Since locations
+                # are ordered from oldest to newest, we can just subtract the first and last timestamps.
+                duration = longest_subset[-1] - longest_subset[0]
+            else:
+                # No streak, so we weren't at the stop for any amount of time.
+                duration = timedelta(seconds=0)
+
+            if duration >= THRESHOLD_TIME:
+                # We were at this stop for long enough, move to it. Since the stops iterated through are relative to the
+                # current stop, we have to add the current stop index to the current index in the loop to get the actual
+                # stop index.
+                # Note: It's possible that the van was at another stop's radius for even longer, but this is not considered
+                # until real-world testing shows this edge case to be important.
+                self.cache.set_current_stop_index(
+                    van_id, (current_stop_index + i + 1) % len(stops)
+                )
+                break
+
+
+def _distance_m(a: Coordinate, b: Coordinate) -> float:
```

Comment:
Why is this called this?

Reply:
`_` - Private function
`distance_` - Distance calculation
`m` - Meters

Reply:
I would just type out meters, and also ensure that if this is the convention we're using, that it proliferates for everyone on the backend